streampark-flink/streampark-flink-kubernetes/src/main/java/org/apache/streampark/archives/FetchArchives.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.streampark.archives;
+
+import org.apache.flink.core.fs.FileStatus;
+import org.apache.flink.core.fs.FileSystem;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.runtime.history.FsJobArchivist;
+import org.apache.flink.runtime.webmonitor.history.ArchivedJson;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+public class FetchArchives {
+
+    public static String fetchArchives(String jobId, String archivePath) {
+        try {
+            Objects.requireNonNull(jobId, "JobId cannot be empty.");
+            Objects.requireNonNull(archivePath, "archivePath cannot be empty.");
+            Path refreshPath = new Path(archivePath);
+            FileSystem refreshFS = refreshPath.getFileSystem();
+            RefreshLocation refreshLocation = new RefreshLocation(refreshPath, refreshFS);
+            Path refreshDir = refreshLocation.getPath();
+            // contents of /:refreshDir
+            FileStatus[] jobArchives;
+            jobArchives = listArchives(refreshLocation.getFs(), refreshDir);
+            Path jobArchivePath = Arrays.stream(jobArchives)
+                .map(FileStatus::getPath)
+                .filter(path -> path.getName().equals(jobId)).findFirst().orElse(null);
+
+            Objects.requireNonNull(jobArchivePath, "Archive Log Path Exception");
+            for (ArchivedJson archive : FsJobArchivist.getArchivedJsons(jobArchivePath)) {
+                String path = archive.getPath();
+                System.out.println(path);
+                if (path.equals("/jobs/overview")) {
+                    String json = archive.getJson();
+                    ObjectMapper objectMapper = new ObjectMapper();
+                    Jobs jobs = objectMapper.readValue(json, Jobs.class);
+                    System.out.println(json);
+                    List<Overview> overviews = jobs.getJobs();
+                    return overviews.get(0).getState();
```

Comment:
If the session mode is used, `jobs.getJobs();` have some jobs.

In general, we should check `overview.getJid` here.

Reply:
I don't quite understand what you mean by

Reply:
> If the session mode is used, `jobs.getJobs();` have some jobs.

That is, `List<Overview> overviews = jobs.getJobs();` will have some element. We should find the correct one based on jobId from overviews.